safe/impact_functions/generic/classified_polygon_people/impact_function.py

```diff
@@ -0,0 +1,411 @@
+                            impact_geometry,
+                            writer):
+        """ Assign different impacted areas with their
+        respective level of impact(Affected, Not Affected, Medium)
+
+        :param f: exposure feature
+        :type f: QgsFeature
+
+        :param hazard_id: id of analyzed hazard
+        :type hazard_id: int
+
+        :param hazard_features: List of hazard features
+        :type hazard_features: {}
+
+        :param unaffected_fields: Unaffected fields in the impact layer
+        :type unaffected_fields: {}
+
+        :param impact_fields: Impacted fields in the impact layer
+        :type impact_fields: {}
+
+        :param unaffected_geometry: untouched geometry by the hazard
+         layer
+        :type unaffected_geometry: QgsGeometry
+
+        :param impact_geometry: touched geometry by the hazard layer
+        :type impact_geometry: QgsGeometry
+
+        :param writer: Object of writing the impact layer
+        :type writer: QgsVectorFileWriter
+
+        """
+
+        # Checking the type of provided hazard using
+        # current flood and earthquake distinguishing
+        # attributes
+        try:
+            hazard_attribute = hazard_features[hazard_id].\
+                attribute('h_zone')
+        except KeyError:
+            try:
+                hazard_attribute = hazard_features[hazard_id].\
+                    attribute('FLOODPRONE')
+            except KeyError:
+                hazard_attribute = None
+
+        f_unaffected = QgsFeature(unaffected_fields)
+        f_impact = QgsFeature(impact_fields)
+
+        if hazard_attribute is not None:
+            f_unaffected.setGeometry(unaffected_geometry)
+            f_unaffected.setAttributes(f.attributes() + [0])
+            f_impact.setGeometry(impact_geometry)
+
+            if hazard_attribute == "Low Hazard Zone":
+                f_impact.setAttributes(f.attributes() + [1])
+
+            elif hazard_attribute == "Medium Hazard Zone":
+                f_impact.setAttributes(f.attributes() + [2])
+
+            elif hazard_attribute == "High Hazard Zone":
+                f_impact.setAttributes(f.attributes() + [3])
+
+            elif hazard_attribute == "YES":
+                f_impact.setAttributes(f.attributes() + [3])
+
+            elif hazard_attribute == "NO":
+                f_impact.setAttributes(f.attributes() + [1])
+
+        else:
+            f_impact.setGeometry(impact_geometry)
+            f_unaffected.setAttributes(f.attributes() + [1])
+            f_impact.setAttributes(f.attributes() + [3])
+
+        writer.addFeature(f_impact)
+        writer.addFeature(f_unaffected)
+
+    def evaluate_affected_people(self):
+        """ Calculate the number of people affected on the area
+        based on the affected area.
+        Currently we assume the population distribution is
+        uniform
+
+        :return:
+        """
+
+        for t, v in self.all_areas_ids.iteritems():
```

Comment:
If possible, don't use single character variable. Except in mathematical expression